### PR TITLE
feat: join group ui

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

This PR adds join group page and integrates it with background services.

Details are below:
- [x] Add join group page
- [x] Add bandada site url
- [x] Support history and notification
- [x] Add event for group join
- [x] Check group membership before joining
- [x] Support user reject event 

## Related Issues

Related to #316 
Blocked by #855

## Screenshots

<details>
  <summary>Expand</summary>

image

image

image

image

image

image
  
</details>

## Manual Testing Steps

N/A

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] Pre-commit and pre-push hook checks are passed
- [x] E2E tests are passed locally
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: packages/app/src/background/services/bandada/BandadaService.ts [hunk 1]
```diff
@@ -47,8 +54,17 @@ export class BandadaService {
     throw new Error(result.message.toString());
   }
 
-  async generateMerkleProof({ groupId, commitment }: IGenerateBandadaMerkleProofArgs): Promise<IMerkleProof> {
-    const response = await fetch(`${API_URL}/groups/${groupId}/members/${commitment}/proof`, {
+  async checkGroupMembership({ identity, groupId }: ICheckBandadaGroupMembershipArgs): Promise<boolean> {
+    const response = await fetch(`${API_URL}/groups/${groupId}/members/${hexToBigint(identity.commitment)}`, {
```
Comment (0xisk) [inline packages/app/src/background/services/bandada/BandadaService.ts:58]: Why do we need this conversion `hexToBigint`?

File: packages/app/src/background/services/credentials/index.ts [hunk 2]
```diff
@@ -59,21 +60,22 @@ export default class VerifiableCredentialsService implements IBackupable {
...
       },
     });
+
     const tabs = await browser.tabs.query({ active: true });
     await Promise.all(
       tabs.map((tab) =>
         browser.tabs
           .sendMessage(tab.id!, {
-            type: EventName.REJECT_VERIFIABLE_CREDENTIAL,
+            type: EventName.USER_REJECT,
```
Comment (0xisk) [inline packages/app/src/background/services/credentials/index.ts:77]: That's a better way of rejecting requests. :ok_hand:

**Review by 0xisk — APPROVED** [review]
Perfect thanks @0xmad!

File: packages/app/src/background/services/bandada/BandadaService.ts (hunk 1 above)
Comment (0xmad) [reply packages/app/src/background/services/bandada/BandadaService.ts:58]: Bandada needs bigint format. Hex values are not supported.

**Pull request merged**